backend/ee/saml/provider/saml.go

```diff
@@ -0,0 +1,167 @@
+			ServiceProviderSLOURL:       fmt.Sprintf("%s/saml/logout", cfg.Saml.Endpoint),
+			SPKeyStore:                  serviceProviderCertStore,
+
+			SignAuthnRequests:       cfg.Saml.Options.SignAuthnRequests,
+			ForceAuthn:              cfg.Saml.Options.ForceLogin,
+			IsPassive:               false,
+			AudienceURI:             cfg.Saml.AudienceUri,
+			ValidateEncryptionCert:  cfg.Saml.Options.ValidateEncryptionCertificate,
+			SkipSignatureValidation: cfg.Saml.Options.SkipSignatureValidation,
+			AllowMissingAttributes:  cfg.Saml.Options.AllowMissingAttributes,
+		},
+	}
+	provider.UseDefaultAttributesIfEmpty()
+
+	return provider, nil
+}
+
+func (sp *BaseSamlProvider) ProvideMetadataAsXml() ([]byte, error) {
+	metadata, err := sp.Service.Metadata()
+	if err != nil {
+		return nil, err
+	}
+
+	// Workaround as the lib is currently marshalling nanoseconds which cannot be used
+	metadata.ValidUntil = time.Now().Add(time.Hour * 24 * 7).Round(time.Millisecond)
+
+	return xml.MarshalIndent(metadata, "", "  ")
+}
+
+func (sp *BaseSamlProvider) GetUserData(assertionInfo *saml2.AssertionInfo) *thirdparty.UserData {
+	firstAssertion := assertionInfo.Assertions[0]
+	assertionValues := assertionInfo.Values
+	attributeMap := &sp.Config.AttributeMap
+
+	emailAddress := assertionValues.Get(attributeMap.Email)
+
+	email := thirdparty.Email{
+		Email:    emailAddress,
+		Verified: false,
+		Primary:  true,
+	}
+
+	expiresIn, _ := time.Parse(time.RFC3339, firstAssertion.Conditions.NotOnOrAfter)
+
+	userData := &thirdparty.UserData{}
+	userData.Emails = append(userData.Emails, email)
+
+	userData.Metadata = &thirdparty.Claims{
+		Issuer:            firstAssertion.Issuer.Value,
+		Subject:           firstAssertion.Subject.NameID.Value,
+		Aud:               sp.Service.AudienceURI,
+		Iat:               float64(assertionInfo.AuthnInstant.Unix()),
+		Exp:               float64(expiresIn.Unix()),
+		Name:              assertionValues.Get(attributeMap.Name),
+		FamilyName:        assertionValues.Get(attributeMap.FamilyName),
+		GivenName:         assertionValues.Get(attributeMap.GivenName),
+		MiddleName:        assertionValues.Get(attributeMap.MiddleName),
+		NickName:          assertionValues.Get(attributeMap.NickName),
+		PreferredUsername: assertionValues.Get(attributeMap.PreferredUsername),
+		Profile:           assertionValues.Get(attributeMap.Profile),
+		Picture:           assertionValues.Get(attributeMap.Picture),
+		Website:           assertionValues.Get(attributeMap.Website),
+		Gender:            assertionValues.Get(attributeMap.Gender),
+		Birthdate:         assertionValues.Get(attributeMap.Birthdate),
+		ZoneInfo:          assertionValues.Get(attributeMap.ZoneInfo),
+		Locale:            assertionValues.Get(attributeMap.Locale),
+		UpdatedAt:         assertionValues.Get(attributeMap.UpdatedAt),
+		Email:             emailAddress,
+		EmailVerified:     assertionValues.Get(attributeMap.EmailVerified) != "" || sp.Config.SkipEmailVerification,
```

Comment:
Ok, I found the mistake in my configuration. Normally auth0 provides its claims with the schema `http://schemas.auth0.com/email_verified`. When you enable the following config-attribute: `mapUnknownClaimsAsIs` auth0 will scratch the `https://schemas.auth0.com`-part on all attributes which are not mapped in the config. 

There are 3 ways how we can solve this:

1. We implement the auth0 defaults with its schema counterpart and document it that way
2. We do not change the auth0 defaults and document that you need to use `mapUnknownClaimsAsIs`
3. We change the algorithm of email_verified to only check if an email is provided. (which is always the case)

Reply:
Another way would be, that I check for both attributes and map the first one which is not empty.

Reply:
I think the first one would be the fastest one to implement and a reasonable safe guard for an default auth0 config. 

It is also documented on the auth0 saml addon:

> passthroughClaimsWithNoMapping (bool): If true (default), for each claim that is not mapped to the [common profile](https://auth0.com/docs/manage-users/user-accounts/user-profiles)
, Auth0 will passthrough those in the output assertion. If false, those claims won't be mapped. Default is true.
mapUnknownClaimsAsIs (bool): if passthroughClaimsWithNoMapping is trueand this is false (default), for each claim that is not mapped to the [common profile](https://auth0.com/docs/manage-users/user-accounts/user-profiles)
 Auth0 will add a prefix http://schema.auth0.com. If true it will passthrough the claim as-is. Default is false.